cpp/src_prims/linalg/svd.cuh

```diff
@@ -212,8 +212,7 @@ template <typename math_t>
 void svdReconstruction(const raft::handle_t &handle, math_t *U, math_t *S,
                        math_t *V, math_t *out, int n_rows, int n_cols, int k,
                        cudaStream_t stream) {
-  std::shared_ptr<raft::mr::device::allocator> allocator =
-    handle.get_device_allocator();
+  auto allocator = handle.get_device_allocator();
 
   const math_t alpha = 1.0, beta = 0.0;
   raft::mr::device::buffer<math_t> SVT(allocator, stream, k * n_cols);
```

Comment:
For this particular example, `raft::mr::device::buffer` is a very lightweight wrapper that helps us allocate device memory on the stack by using RAII approach. In which case, it has no reason to use something as complicated as the entire handle.

For other examples, the distinguishing factor between passing down handle to child functions is decided on a similar need-basis. If child functions need any information from the handle, we pass it down, and otherwise not.